plugins/modules/atlas_ldap_user.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020 T-Systems MMS
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "0.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: atlas_ldap_user
+short_description: Manage database users in Atlas
+description:
+   - The atlas_users module lets you create, modify and delete the database users in your cluster.
+   - Each user has a set of roles that provide access to the project's databases.
+   - A user's roles apply to all the clusters in the project
+   - if two clusters have a products database and a user has a role granting read access on the products database,
+   - the user has that access on both clusters.
+   - L(API Documentation,https://docs.atlas.mongodb.com/reference/api/database-users/)
+author: "Martin Schurz (@schurzi)"
+options:
+  apiUsername:
+    description:
+      - The username for use in authentication with the Atlas API.
+      - Can use API users and tokens (public key is username)
+    type: str
+    required: True
+  apiPassword:
+    description:
+      - The password for use in authentication with the Atlas API.
+      - Can use API users and tokens (private key is password)
+    type: str
+    required: True
+  state:
+    description:
+      - State of the ressource.
+    choices: [ "present", "absent" ]
+    default: present
+    type: str
+  groupId:
+    description:
+      - Unique identifier for the Atlas project.
+    type: str
+    required: True
+  databaseName:
```

Comment:
this should be hardcoded to external, or am I understanding the documentation wrong?

Reply:
If you enable user authorization with LDAP, you can create LDAP groups on the admin database by mapping LDAP groups to MongoDB roles on your Atlas databases. To use LDAP groups effectively, create additional projects within Atlas to control access to specific deployments in your organization, such as creating separate Atlas projects for development and production environments. You can then map an LDAP group to a role in the Atlas project to provide access to the desired deployment.

Reply:
ahh ok, so I misunderstood. Only the connecting client needs to specify `$external`